docs/architecture.rst

```diff
@@ -3,132 +3,37 @@
-models. This ensures that every compile is repeatable and allows collaboration because all changes *have* to be committed.
-
-
-Agent modes
------------
-
-The Inmanta agent performs all changes in the infrastructure. Either the orchestration server starts an agents or
-an agent is deployed as a separate process.
-
- * **agentless**: Autostarted agents allow for an agentless mode: no explicit agents need to be started. When the agent needs to make changes on machine/vm it can make the changes over remote over ssh. Autostarted agents are controlled by using :inmanta:entity:`std::AgentConfig`. :inmanta:entity:`ip::Host` and subclasses can automatically configure an agent with the `remote_agent` attribute.
- * **external agent**: External agent processes need explicit configuration to connect to the orchestration server. The aws and openstack modules use the platform module to generate a user_data bootscript for virtual machines to install an agent and connect to the orchestration server. The `install_agent` boolean controls this option.
 
 
 Resource deployment
 -------------------
 
-The agent is responsible for:
-
- * repair the infrastructure at regular intervals
- * change the infrastructure at regular intervals
- * enforce desired state when the server requests it
-
-Repair
-******
-At regular intervals the agent verifies that the current state of all resources it manages matches the desired state provided by the orchestration server. For a repair the agent verifies all resources, even if the last known current state already matches the desired state. In the current release all deploys are done through a repair and run by default every 600 seconds. This is controlled with :inmanta.config:option:`config.agent-repair-interval`, when this option is set to 0 no repairs are performed.
+The scheduler deploys resources (through the executors and the smart adapters) in 5 ways:
 
-Deploy changes
-**************
-For very large infrastructures or infrastructure that is too slow (for example network devices with underpowered control planes or thousands of managed resources) a repair cannot run often. For example, only once a  week. When this is the case, the agent can deploy only known changes (based on the previous deployed state cached by the orchestration server). This interval is controlled by :inmanta.config:option:`config.agent-deploy-interval`. This interval should be a lot shorter than :inmanta.config:option:`config.agent-repair-interval`
+1. Deployment of changes: when the intent (desired state) on the server changes, the scheduler will instruct the executors to update the infrastructure using the smart adapters.
+2. Self-healing: when a resource has not been verified by the server for some time, it will verify the intent still holds. If it doesn't the intent is restored. This ensure that the managed infrastructure doesn't deteriorate over time.
```

Comment:
```suggestion
2. Self-healing: when a resource has not been verified by the server for some time, it will verify the intent still holds. If it doesn't the intent is restored. This ensures that the managed infrastructure doesn't deteriorate over time.
```